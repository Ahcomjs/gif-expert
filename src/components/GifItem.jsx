import PropTypes from 'prop-types';

export const GifItem = ({ title, url }) => {
    return (
        <div className="gif-item-container-card">
            <img src={url} alt={title} className="gif-card-image" />
            <p className="gif-title">{title}</p>
        </div>
    );
};


GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}






