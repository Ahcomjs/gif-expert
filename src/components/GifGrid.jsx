import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGits";
import { PropTypes } from 'prop-types';


export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            {
              isLoading && (<h2>Loading...</h2>)
            }
            <div className="card-grid">
                {
                    images.map(image => (
                        <GifItem
                            key={image.id}
                            {...image}
                        />
                    ))
                }
            </div>
        </>
    )
}


GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}