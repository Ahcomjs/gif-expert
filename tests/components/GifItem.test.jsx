import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Test GifItem', () => {

    const title = "Saitama";
    const url = "https://example.com/";


    test('Should match with the Snapshot', () => {

        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();

    });

    test('Should show the image with the URL and the ATL', () => {

        render(<GifItem title={title} url={url} />);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);

    });

    test('Should validate if the title exists', ()=>{
        render(<GifItem title={title} url={url} />)

        expect(screen.getByText(title)).toBeTruthy();
    });

});