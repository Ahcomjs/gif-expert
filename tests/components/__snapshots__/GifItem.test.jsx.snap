// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifItem Should match with the Snapshot 1`] = `
<div>
  <div
    class="gif-item-container-card"
  >
    <img
      alt="Saitama"
      class="gif-card-image"
      src="https://example.com/"
    />
    <p
      class="gif-title"
    >
      Saitama
    </p>
  </div>
</div>
`;
